plugins/parsers/opentsdb/parser.go

```diff
@@ -0,0 +1,127 @@
+package opentsdb
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+// Parser encapsulates a OpenTSDB Parser.
+type Parser struct {
+	DefaultTags map[string]string `toml:"-"`
+}
+
+func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
+	var metrics []telegraf.Metric
+
+	scanner := bufio.NewScanner(bytes.NewReader(buf))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// delete LF and CR
+		if line[len(line)-1] == '\n' {
+			line = line[:len(line)-1]
+		}
+		if line[len(line)-1] == '\r' {
+			line = line[:len(line)-1]
+		}
```

Comment:
Trimming at the start of the string bothers me a bit so I've used `TrimRight` instead

Reply:
Do you expect any spaces at the start of the line?

Reply:
The line is something like "put .....\r\n".

If there are spaces at the start, it is invalid, and should be rejected - at least to me. Trimming it would lead the line to be accepted instead.

Reply:
Understood. Can you please add this comment to the code?!